File: Madera/Madera/Models/Devis.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Madera.Models
{

    public class Devis
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int ID { get; set; }
        public string LibelleDevis { get; set; }
        public string EtatDevis { get; set; }
        [ForeignKey("PlanID")]
        public int PlanID { get; set; }
        public DateTime DateDebutDevis { get; set; }
        [Column(TypeName = "decimal(10, 2)")]
        public decimal PrixTotalHtDevis { get; set; }
        [Column(TypeName = "decimal(10, 2)")]
        public decimal PrixTotalTtcDevis { get; set; }
        [Column(TypeName = "decimal(10, 2)")]
        public decimal MargeCommercialDevis { get; set; }
        [Column(TypeName = "decimal(10, 2)")]
        public decimal MargeEntrepriseDevis { get; set; }
        public int IdUtilisateurCreation { get; set; }
        public DateTime DateCreation { get; set; }
        public int IdUtilisateurModification { get; set; }
        public DateTime DateModification { get; set; }
        public DateTime? DateArchivage { get; set; }

        public virtual Plan Plan { get; set; }
    }
}

File: Madera/Madera/ClientApp/src/app/models/Devis.ts
export class Devis {

    id?: number;
    LibelleDevis: string;
    EtatDevis: string;
    planID: number;
    DebutDateDevis: Date;
    PrixTotalHtDevis: number;
    PrixTotalTtcDevis: number;
    MargeCommercialDevis: number;
    MargeEntrepriseDevis: number;
    IdUtilisateurCreation: number;
    DateCreation: Date;
    IdUtilisateurModification: number;
    DateModification: Date;
    DateArchivage: Date;

}

export class SearchingDevis extends Devis {
    id?: number;
    LibelleDevis : string;
    DateCreation : Date;

  }
